### PR TITLE
Add prefix of `com.openfaas.cloud.` to labels

### LINKED ISSUE #155: Prefix openfaas-cloud labels with com.openfaas.cloud.
## Expected Behaviour

We should be able to filter out our own labels vs. user labels.

Prefix all labels with `com.openfaas.cloud.`

i.e.

```
com.openfaas.cloud.git-owner: alexellis
```

## Current Behaviour

The labels are not namespaced or prefixed, so if we wanted to filter we'd have to keep a whole list of all the variants.

## Solution

While working on this solution, let's also make all labels lower-case for the label value for consistency.

* Prefix openfaas-cloud labels with "com.openfaas.cloud."
* Make all labels lowercase
* Test thoroughly end-to-end including the dashboard UI

### PR DESCRIPTION
**Pull request by martindekov**
Updating the labels that buildshiprun applies when creating
functions by lowercasing them (except git-SHA) and appending
`com.openfaas.cloud.` prefix in front of the label also
changed the label handling in garbage-collect, list-functions
and index.html in overview dashboard

Signed-off-by: Martin Dekov (VMware) <mdekov@vmware.com>

## Description

In `buildshiprun` we apply labels to the function and I modified the labels from `Git-Repo`->`com.openfaas.cloud.git-repo` also small alternation in `list-functions` `overview` and `garbage-collect` to handle the change

## How Has This Been Tested?

Tested end to end the following ways:

Deployed dashboard with the following `pretty_url` configuration and of-router on the following way
```
...
      pretty_url: "http://user.team-serverless.xyz/function"
      query_pretty_url: "true"
...
```
the output of `https://md-gw.team-serverless.xyz/system/function/martindekov-gocloudfunc` is
```
{"name":"martindekov-gocloudfunc","image":"md-reg.team-serverless.xyz/martindekov/gocloudfunc-gocloudfunc:latest-4aa81b8","invocationCount":0,"replicas":1,"envProcess":"","availableReplicas":1,"labels":{"com.openfaas.cloud.app":"martindekov-gocloudfunc","com.openfaas.cloud.faas_function":"martindekov-gocloudfunc","com.openfaas.cloud.git-SHA":"4aa81b8e859b7487fbfbe6518b49ff4bc0384d09","com.openfaas.cloud.git-cloud":"1","com.openfaas.cloud.git-deploytime":"1536656802","com.openfaas.cloud.git-owner":"martindekov","com.openfaas.cloud.git-repo":"gocloudfunc","com.openfaas.function":"martindekov-gocloudfunc","function":"true"}}
```
which shows that the prefixes are applied in the intended places

Tested garbage collect by removing the OF installation from repo resulting in removed function in cloud.

Tested the logs just in case they work with the change. 

Tested the `Repo` `Endpoint` `Logs` with my case sensitive github account everything referenced the right thing and the `system-list-functions` output is:
```
{"name":"martindekovwat-goforfun","image":"md-reg.team-serverless.xyz/martindekovwat/exmpl-goforfun:latest-e2d1646","invocationCount":0,"replicas":1,"envProcess":"","availableReplicas":1,"labels":{"com.openfaas.cloud.app":"martindekovwat-goforfun","com.openfaas.cloud.faas_function":"martindekovwat-goforfun","com.openfaas.cloud.git-SHA":"e2d1646c72570a218c9c1ce037fd788a70fdc5e7","com.openfaas.cloud.git-cloud":"1","com.openfaas.cloud.git-deploytime":"1536662023","com.openfaas.cloud.git-owner":"MartinDekovWat","com.openfaas.cloud.git-repo":"exmpl","com.openfaas.function":"martindekovwat-goforfun","function":"true"}}
```

Closes #155 

Note: The functions need to be redeployed for the change to work

## Checklist:

I have:

- [ ] updated the documentation if required
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: buildshiprun/handler.go [hunk 1]
```diff
@@ -162,15 +162,15 @@ func Handle(req []byte) string {
...
-				"Git-Cloud":      "1",
-				"Git-Owner":      event.Owner,
-				"Git-Repo":       event.Repository,
-				"Git-DeployTime": strconv.FormatInt(time.Now().Unix(), 10), //Unix Epoch string
-				"Git-SHA":        event.SHA,
-				"faas_function":  serviceValue,
-				"app":            serviceValue,
-				"com.openfaas.scale.min": scalingMinLimit,
-				"com.openfaas.scale.max": scalingMaxLimit,
+				"com.openfaas.cloud.git-cloud":      "1",
```
Comment (alexellis) [inline buildshiprun/handler.go]: Suggest we put this prefix into a const and store it in the SDK package?

**Comment by alexellis** [comment]
Please can you help me deploy this fix?

* bump all versions in stack.yml and let me know which functions I need to rebuild and redeploy.
* write a script or piece of code that I can use to migrate the existing users workloads in the live environment - if I deploy this now all their functions will disappear right? I don't mind if this is a throw-away script or hack - just something to support our live users.

In effect you're saying -

* for each deployment
* if it has a label of x
* get that label and apply it again with a new prefix of `com.openfaas.cloud`

Does that make sense? Maybe we can even use the OpenFaaS REST API and a simple piece of Python code in a loop?

Alex

File: buildshiprun/handler.go (hunk 1 above)
Comment (martindekov) [reply buildshiprun/handler.go]: It's a good idea. Should I put only `com.openfaas.cloud` as single const or the whole `com.openfaas.cloud.git-cloud` and add consts for all the variables like `const gitRepo = `com.openfaas.cloud.git-repo`

**Comment by martindekov** [comment]
I will need to think about the script you mentioned other things are easy, also I need to update the PR with ken's changes and test that

**Comment by alexellis** [comment]
This is blocked pending the migration script/code.

**Comment by alexellis** [comment]
Derek add label: blocked

File: buildshiprun/handler.go (hunk 1 above)
Comment (alexellis) [reply buildshiprun/handler.go]: @martindekov cc @bartsmykla please raise a quick PR to the SDK component to add a const for this prefix. I'll then do a merge / release and you can vendor it here.

Comment (martindekov) [reply buildshiprun/handler.go]: Ok I've already done that in the unpushed changes But will make it into another PR

**Comment by alexellis** [comment]
@martindekov please share your migration gist here and on the issue.

**Comment by martindekov** [comment]
Redid testing due to the many changes I've made.
Re-pushed function and `buildshiprun` succesfully applied new labels
output of `https://md-gw.team-serverless.xyz/function/list-functions?user=martindekov`
is
```
[{"name":"martindekov-gocloudfunc","image":"martindekov-gocloudfunc-gocloudfunc:latest-313019c","invocationCount":3,"replicas":1,"labels":{"app":"martindekov-gocloudfunc","com.openfaas.cloud.git-SHA":"313019c8340e5c57a5bbc7478c34cb1233602605","com.openfaas.cloud.git-cloud":"1","com.openfaas.cloud.git-deploytime":"1537273715","com.openfaas.cloud.git-owner":"martindekov","com.openfaas.cloud.git-repo":"gocloudfunc","com.openfaas.function":"martindekov-gocloudfunc","com.openfaas.scale.max":"4","com.openfaas.scale.min":"1","faas_function":"martindekov-gocloudfunc","function":"true"}}]
```
Which shows that the labels are applied and `list-function` shows them also `garbage-collect` sucessfully removes function on uninstall event.
TO DO: 
* test dashboard

**Comment by bartsmykla** [comment]
Looks good to me

File: dashboard/client/src/api/functionsApi.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ class FunctionsApi {
         endpoint = this.baseURL + '/function/' + item.name;
       }
 
-      let shortSha = item.labels['Git-SHA'];
+      let shortSha = item.labels['com.openfaas.cloud.git-SHA'];
```
Comment (alexellis) [inline dashboard/client/src/api/functionsApi.js]: I'm going to guess we should make `SHA` lower too so that it matches everything else?

Comment (martindekov) [reply dashboard/client/src/api/functionsApi.js]: Ok

**Comment by alexellis** [comment]
@martindekov great work. Please raise a follow-up PR for my comment above and also update the gist to be called main.go.

**Pull request merged**